src/components/TaxPayer/SpouseAndDependent.test.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import userEvent from '@testing-library/user-event'
+
+import { SpouseInfo } from './SpouseAndDependent'
+import { store } from '../../redux/store'
+
+afterEach(async () => {
+  await waitFor(() => localStorage.clear())
+  jest.resetAllMocks()
+})
+
+describe('SpouseInfo', () => {
+  it('renders an `Add` button when no spouse has been added', () => {
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
+    )
+    // initial state has an add button
+    screen.getByRole('button', {
+      name: /Add/
+    })
+
+    // initial state does not have any forms or labels
+    const firstNameLabel = screen.queryByLabelText('First Name and Initial')
+    expect(firstNameLabel).not.toBeInTheDocument()
+
+    const lastNameLabel = screen.queryByLabelText('Last Name')
+    expect(lastNameLabel).not.toBeInTheDocument()
+
+    const ssnLabel = screen.queryByLabelText('SSN / TIN')
+    expect(ssnLabel).not.toBeInTheDocument()
+  })
+  it('renders form elements when `Add` button is clicked', () => {
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
+    )
+
+    const addButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    fireEvent.click(addButton)
+
+    // forms and labels appear after clicking the add button
+    screen.getByText('First Name and Initial')
+    screen.getByText('Last Name')
+    screen.getByText('SSN / TIN')
+
+    screen.getByRole('button', {
+      name: /Add/
+    })
+
+    screen.getByRole('button', {
+      name: /Close/
+    })
+  })
+  it('saves and edits a spouse', async () => {
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
+    )
+
+    const addButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    fireEvent.click(addButton)
+
+    const createButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    screen.getByRole('button', {
+      name: /Close/
+    })
+
+    // get all three text inputs
+    const inputs = screen.getAllByRole('textbox')
+    const [firstNameInput, lastNameInput, ssidInput] = inputs
+
+    // add values for each input
+    userEvent.type(firstNameInput, 'Sally K')
+    userEvent.type(lastNameInput, 'Ride')
+    fireEvent.change(ssidInput, { target: { value: '123456789' } })
```

Comment:
I'm not sure why, probably due to nested elements, but `userEvent.type` was not working for ssid, so I opted for `fireEvent` instead.